### ISSUE TITLE
Top Of Book

### ISSUE THREAD

**Issue by gRittyKitty** [coinbase/gdax-tt#139]
I really like what you have done so far. Thank you. I'm using the tutorials as a base and trying to expand from there. Although I've programmed before (Python most recently), I'm new to typescript/javascript. Perhaps this is not the forum to ask for some help but I don't see a forum for help -- if there is, please point me to it.

My specific problem is that I'd like to simply see top of book as it updates. I am trying to use the LiveOrderBook example #2: https://github.com/coinbase/gdax-tt/blob/master/tutorials/t002_liveOrderbook.ts

I've added to the FeedFactory streamhandler (please instruct me if I have used the terminology incorrectly) a new book.on message handler from the LiveOrderBook:
```
   book.on('data',() => {`
             console.log(`${book.highestBid.toFixed(3)}`);
   });
```

This does not work. I get "undefined". When I look at the API docs, it appears that book does have the accessor highestBid (https://coinbase.github.io/gdax-tt/apiref/classes/_src_lib_bookbuilder_.bookbuilder.html)

I've also tried using the state:
```
const state = book.state();
console.log(`highest bid:  ${state.highestBid}`);
```

In addition, should I be using the message "data" to check for top of book updates? or should I use "ticker"? Or something else?

So I feel like I'm not reading the API docs correctly. Please advise on my questions and, perhaps, how I ought to be accessing the top of book rather than the directions I have taken.

edit: fixed formatting problem.

**Comment by gRittyKitty** [comment]
I (finally) figured this out. For those that are interested, here is how to solve the problem:
```
const state = ltc_book.state();
const bid: PriceLevel = state.bids[0];
console.log('bid : ', bid.price);
console.log(`here : ${padfloat(bid.price, 7, 2)}`);
```
I've included the raw console.log of bid.price, but it won't work they way that you want it to as it is a "BigJS" number. 

output looks like this:

```
bid :  BigNumber { s: 1, e: 2, c: [ 148, 62000000000000 ] }
here :  148.62
```

Note that I've hardcoded the index to the bids to 0. I based this on code in 
utils/printer.ts

https://github.com/coinbase/gdax-tt/blob/master/src/utils/printers.ts
